### DISCUSSION TITLE
Generic component field.onChange/value shows type as PathValue<TFieldValues, TPath>

### DISCUSSION THREAD

**Discussion by tparvi** [react-hook-form/react-hook-form#10615] [Q&A]
Codesandbox: https://codesandbox.io/s/rhf-7-45-1-generic-sdx8hz

We have a generic components that use `useController` and you should only be able to use it with fields that are `number`. Everything else works except that the `field.value`, `useWatch` and `field.onChange` are not correctly typed. In reality I am not using `useWatch` to monitor the value but I wanted to see if the problem is there too.

As you can see in the following the types are incorrect. The type should be `number` but TS shows it is `PathValue<TFieldValues, TPath>`.

![image](https://github.com/react-hook-form/react-hook-form/assets/871053/4861220a-af5d-478e-9659-4909af5f8e9a)
![image](https://github.com/react-hook-form/react-hook-form/assets/871053/edd80f56-b89e-4ef2-8b12-110b44bb9f6d)

I think the problem is with `UseControllerProps` but I cannot figure out how do I change my component to let TS know that the value, onChange and useWatch all use `number` while keeping the constraint that component can only be used with fields of type `number`.

**Comment by araphiel-chwy** [comment]
Definitely worth exploration. I hope [`const` generics](https://github.com/microsoft/TypeScript/pull/51865) can be used to sort this one out.

As things stand today, I'm keeping my RFH wrapper as generic [as possible](https://codesandbox.io/s/rhf-7-45-1-generic-forked-496zt9?file=/src/App.tsx) and using that as a base for a form specific component (pass your form typed `control` as a prop) to handle specific logic needs.
